src/Go.CodeAnalysis/Text/StringSnapshot.cs

```diff
@@ -1,7 +1,13 @@
 ﻿namespace Go.CodeAnalysis.Text
 {
+    /// <summary>
+    /// A text segment based on a concrete string.
```

Comment:
nit: a ‘text snapshot’. I’ve been using ‘Segment’ in place of ‘Span’, so when you see ‘Segment’, it indicates a subspan in a document, and snapshot is a complete document.

Reply:
Hmmm then SnapshotSegment can be confusing based on this definition.